service/controller/app/resources.go

```diff
@@ -262,13 +262,15 @@ func newAppResources(config appResourcesConfig) ([]resource.Interface, error) {
 	}
 
 	resources := []resource.Interface{
+		// validationResource checks CRs for validation errors and sets the CR status.
+		validationResource,
```

Comment:
Moving the validation first so its before the clients resource. As the clients resource needs to use the kubeconfig secret.